src/plugins/plotly/PlotlyDiagram.vue

```diff
@@ -153,45 +159,91 @@ const MyComponent = defineComponent({
     await this.getVizDetails()
     // only continue if we are on a real page and not the file browser
     if (this.thumbnail) return
-
     try {
       if (this.vizDetails.datasets) await this.prepareData()
       if (this.vizDetails.traces) this.traces = this.vizDetails.traces
+
       // merge user-supplied layout with SimWrapper layout defaults
       if (this.vizDetails.layout) this.mergeLayouts()
-
       if (this.vizDetails.fixedRatio) {
         this.vizDetails.layout.xaxis = Object.assign(this.vizDetails.layout.xaxis, {
           constrain: 'domain',
         })
-
         this.vizDetails.layout.yaxis = Object.assign(this.vizDetails.layout.yaxis, {
           constrain: 'domain',
           scaleanchor: 'x',
           scaleration: 1,
         })
       }
-
       // Backwards compatiblity with the older "dropdownMenu" option
       if (this.vizDetails.dropdownMenu) this.vizDetails.interactive = 'dropdown'
-
       // create interactive elements
       if (this.vizDetails.interactive) this.createMenus(this.vizDetails.interactive)
+
+      this.calculateAxis()
     } catch (err) {
       const e = err as any
       console.error({ e })
       this.loadingText = '' + e
     }
-
     this.updateTheme()
     window.addEventListener('resize', this.changeDimensions)
+    this.layout.margin = { r: 20, t: 0, b: 0, l: 30, pad: 0 }
+    console.log(this.layout)
   },
 
   beforeDestroy() {
     window.removeEventListener('resize', this.changeDimensions)
   },
 
   methods: {
+    /**
+     * Calculates the axis ranges for the x-axis and y-axis based on the data in the 'traces' array.
+     * It iterates through each trace and updates the 'maxXValue', 'maxYValue', 'minYValue', and 'minXValue'
+     * based on the maximum and minimum values found in the 'x' and 'y' arrays of each trace.
+     */
+    calculateAxis() {
+      for (let i = 0; i < this.traces.length; i++) {
+        // Update the 'maxXValue' if the maximum value in the 'x' array of the current trace is greater than the current 'maxXValue'.
+        if (Math.max(...this.traces[i].x) >= this.maxXValue) {
```

Comment:
This is really inefficient! You are calculating the max over the entire array twice for no reason! Use a local variable